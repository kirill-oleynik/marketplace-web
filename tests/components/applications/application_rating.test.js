const React = require('react');
const { shallow } = require('enzyme');
const toJSON = require('enzyme-to-json').default;
const {
  ApplicationRating
} = require('../../../app/components/applications/application_rating');

describe('#render', () => {
  test('it renders correctly', () => {
    const component = shallow(
      <ApplicationRating
        t={(translation) => translation}
        data={{
          average: 0,
          total: 0,
          votes: {
            1: 0,
            2: 0,
            3: 0,
            4: 0,
            5: 0
          }
        }}
      />
    );

    expect(
      toJSON(component)
    ).toMatchSnapshot();
  });
});
