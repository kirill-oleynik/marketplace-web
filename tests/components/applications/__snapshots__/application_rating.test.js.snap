// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render it renders correctly 1`] = `
<div
  className="mr-70"
>
  <p
    className="font-16 font-700 mb-20"
  >
    appProfile.ratings.avarage
  </p>
  <div
    className="d-flex mb-30"
  >
    <div
      className="mr-25"
    >
      <p
        className="font-54 lh-1-2 in-black-050 mb-10"
      >
        0.0
      </p>
      <RatingPreview
        average={0}
        className="rating--sm"
        t={[Function]}
        total={0}
      />
    </div>
    <div>
      <table
        cellPadding="0"
        cellSpacing="0"
      >
        <tbody>
          <tr
            className="rating-marks"
          >
            <td>
              <span
                className="font-12 mr-20"
              >
                appProfile.ratings.votesPoints
              </span>
            </td>
            <td>
              <div
                className="relative"
              >
                <div
                  className="rating-marks__line"
                />
                <div
                  className="rating-marks__line-value"
                  style={
                    Object {
                      "width": "NaN%",
                    }
                  }
                />
              </div>
            </td>
            <td>
              <div
                className="rating-marks__value"
              >
                0
              </div>
            </td>
          </tr>
          <tr
            className="rating-marks"
          >
            <td>
              <span
                className="font-12 mr-20"
              >
                appProfile.ratings.votesPoints
              </span>
            </td>
            <td>
              <div
                className="relative"
              >
                <div
                  className="rating-marks__line"
                />
                <div
                  className="rating-marks__line-value"
                  style={
                    Object {
                      "width": "NaN%",
                    }
                  }
                />
              </div>
            </td>
            <td>
              <div
                className="rating-marks__value"
              >
                0
              </div>
            </td>
          </tr>
          <tr
            className="rating-marks"
          >
            <td>
              <span
                className="font-12 mr-20"
              >
                appProfile.ratings.votesPoints
              </span>
            </td>
            <td>
              <div
                className="relative"
              >
                <div
                  className="rating-marks__line"
                />
                <div
                  className="rating-marks__line-value"
                  style={
                    Object {
                      "width": "NaN%",
                    }
                  }
                />
              </div>
            </td>
            <td>
              <div
                className="rating-marks__value"
              >
                0
              </div>
            </td>
          </tr>
          <tr
            className="rating-marks"
          >
            <td>
              <span
                className="font-12 mr-20"
              >
                appProfile.ratings.votesPoints
              </span>
            </td>
            <td>
              <div
                className="relative"
              >
                <div
                  className="rating-marks__line"
                />
                <div
                  className="rating-marks__line-value"
                  style={
                    Object {
                      "width": "NaN%",
                    }
                  }
                />
              </div>
            </td>
            <td>
              <div
                className="rating-marks__value"
              >
                0
              </div>
            </td>
          </tr>
          <tr
            className="rating-marks"
          >
            <td>
              <span
                className="font-12 mr-20"
              >
                appProfile.ratings.votesPoints
              </span>
            </td>
            <td>
              <div
                className="relative"
              >
                <div
                  className="rating-marks__line"
                />
                <div
                  className="rating-marks__line-value"
                  style={
                    Object {
                      "width": "NaN%",
                    }
                  }
                />
              </div>
            </td>
            <td>
              <div
                className="rating-marks__value"
              >
                0
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
